modules/mod_ginger_base/support/ginger_search.erl

```diff
@@ -10,20 +10,37 @@
 -include_lib("zotonic.hrl").
 
 -define(GINGER_SEARCH_PIVOT, ginger_search).
+-define(GINGER_SEARCH_ARGUMENTS, [
+    anykeyword,
+    cat_exclude_defaults,
+    cat_exclude_unfindable,
+    custompivots,
+    filters,
+    has_geo,
+    is_findable,
+    keyword,
+    ongoing_on_date
+]).
 
-%% @doc Supports all the usual query model arguments, adds default excludes.
-search_query(#search_query{search={ginger_search, Args}}, Context) ->
+%% @doc Transform custom Ginger search arguments to Zotonic search arguments and
+%%      forward the search query.
+%%      Supports all the usual query model arguments, adds default excludes.
+search_query(#search_query{search = {ginger_search, Args}, offsetlimit = Offset}, Context) ->
 
     % This is a special use case that needs a better solution in Zotonic
-    case z_context:get_q(filters, Context) of
+    Args1 = case z_context:get_q(filters, Context) of
         undefined ->
-            Args1 = Args;
+            Args;
         Filters ->
-            Args1 = lists:append([Args, [{filters, Filters}]])
+            lists:append([Args, [{filters, Filters}]])
     end,
 
     QueryArgs = merge_ginger_args(Args1, Context),
-    search_query:search(QueryArgs, Context).
```

Comment:
This blocked the chain of observers, which is almost always a bad thing as it makes the code [closed for extension](https://en.wikipedia.org/wiki/Open/closed_principle).